File: core/src/main/java/com/palomiklo/streampulse/blueprint/Event.java
package com.palomiklo.streampulse.blueprint;

import static com.palomiklo.streampulse.serialization.Mapper.getMapper;

import java.util.UUID;

import static com.palomiklo.streampulse.util.Wrap.wrap;

public record Event(String id, String data) {

    public Event(String id, String data) {
        this.id = id;
        this.data = data;
    }

    public Event(UUID id, String data) {
        this(id.toString(), data);
    }

    public static String serialize(Object data) {
        return wrap(() -> "data: " + getMapper().writeValueAsString(data) + "\n\n", "Failed to serialize event: ", () -> "{}");
    }
}

File: core/src/main/java/com/palomiklo/streampulse/blueprint/Event.kt
package com.palomiklo.streampulse.blueprint

import com.palomiklo.streampulse.serialization.Mapper
import com.palomiklo.streampulse.util.Wrap.wrap
import java.util.UUID

@JvmRecord
data class Event(val id: String, val data: String) {
    constructor(id: UUID, data: String) : this(id.toString(), data)

    companion object {
        fun serialize(data: Event): String {
            return wrap(
                { "data: " + Mapper.mapper.writeValueAsString(data) + "\n\n" },
                "Failed to serialize event: ")
        }
    }
}
